salt/metalk8s/archives/unmounted.sls

```
@@ -0,0 +1,31 @@
+{# Remove mounted ISOs which were removed from configuration #}
+{%- set desired = salt.metalk8s.get_archives() %}
+{%- set available = salt.metalk8s.get_mounted_archives() %}
+{%- set ns = namespace(found_archives_to_remove=false) %}
+
+{%- for env, archive in available.items() %}
+  {%- if env not in desired %}
+    {%- set ns.found_archives_to_remove = true %}
+
+Unmount '{{ archive.iso }}' from '{{ archive.path }}':
+  mount.unmounted:
+    - name: {{ archive.path }}
+    - device: {{ archive.iso }}
+    - persist: true
+
+Remove archive mountpoint '{{ archive.path }}':
+  file.absent:
+    - name: {{ archive.path }}
+    - unless:
+      - fun: file.path_exists_glob
+        path: {{ archive.path }}/*
```

So in this case we just silently ignore it and we do not remove the directory ? :thinking: 
Ok fine

---

That was your suggestion, I just implemented it :innocent: But it makes sense to tell the user about it, I'll add a state for this.

---

Oh, indeed :smile: